app/controllers/media_objects_controller.rb

```diff
@@ -54,6 +55,26 @@ def confirm_remove
     raise CanCan::AccessDenied unless Array(params[:id]).any? { |id| current_ability.can? :destroy, MediaObject.find(id) }
   end
 
+  def intercom_collections
+    respond_to do |format|
+      format.json do
+        render json: Avalon::Intercom.user_collections(user_key).to_json
+      end
+    end
+  end
+
+  def intercom_push
+    result = Avalon::Intercom.push_media_object(@media_object, params[:collection_id], params[:include_structure] == 'true')
```

Comment:
It looks like there isn't any authorization done to ensure that the user posting to this action is allowed to push a media object to `params[:collection_id]`.  Maybe all that would be necessary is for the user_key to be passed along to the other avalon for it to make the authZ check.

Also same concern as above that only users authorized to push to other avalons can actually execute this.

Reply:
This user auth check on the target collection happens in the intercom lib.

Reply:
I looked again but couldn't find it.  Could you point me to where this is happening?

Reply:
https://github.com/avalonmediasystem/avalon/commit/9d05b503d0cf4ce755ba88e728f0a02b59752ba3#diff-ee2a2474f4addd4b9834b6dcd21545bbR35

Reply:
I think we should add an ability for intercom_push that we check here (and can also check in the view to determine if the UI for intercom pushes should be shown) and raise CanCan::AccessDenied if not authorized.  

We'll also need to check that the `collection_id` is one that was returned by intercom.  I think this would best be done in `Avalon::Intercom`.  I'm wondering if we should make `Avalon::Intercom` be a class that can be initialized with a user and avalon and then it can memoize the user_collections and maybe avoid duplicate api calls.